File: algo/KokoEatingBananas.java
class KokoEatingBananas {
    // Approach 2: Brute Force
    // Time Complexity: O(N*M)
    // Space Complexity: O(1)
    public int minEatingSpeed2(int[] piles, int h) {
        int max = 0;
        int min = 1;
        for (int num : piles) {
            max = Math.max(num, max);
        }

        for (int speed = min; speed < max; speed++) {
            int currH = 0;
            for (int pile : piles) {
                currH += Math.ceil((double) pile / speed);
                if (currH > h) {
                    break;
                }
            }
            if (currH <= h) {
                return speed;
            }
        }
        return max;
    }
}

File: algo/KokoEatingBananas.cpp
class Solution
{
public:
    int minEatingSpeed(vector<int> &piles, int h)
    {
        int low = 1;
        int high;
        for (int i = 0; i < piles.size(); i++)
        {
            high = max(high, piles[i]);
        }
        int result = high;
        while (low <= high)
        {
            int mid = low + (high - low) / 2;
            long int hours = 0;
            for (int i = 0; i < piles.size(); i++)
            {
                hours += ceil((double)piles[i] / mid);
            }
            if (hours <= h)
            {
                result = min(result, mid);
                high = mid - 1;
            }
            else
            {
                low = mid + 1;
            }
        }
        return result;
    }
};